### PR TITLE
Update dex config to enable gRPC API on dev

### PR DESCRIPTION
**Pull request by floreks**
**What this PR does / why we need it**:
We need to expose dex API to be able to dynamically create users for e2e tests. This will only be done for `dev.kubermatic.io`. In the future when we will be able to start from scratch a clean environment this won't be needed.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Related to https://github.com/kubermatic/dashboard-v2/issues/958

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by floreks** [comment]
https://github.com/kubermatic/secrets/pull/216 should be merged first.
/hold

File: config/kubermatic/templates/dex-grpc-secret.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: dex-grpc-certs
+type: Opaque
+data: {{ if .Values.dex.grpc }}
+  ca.crt: "{{ .Values.dex.grpc.certs.caCert }}"
+  server.crt: "{{ .Values.dex.grpc.certs.serverCert }}"
+  server.key: "{{ .Values.dex.grpc.certs.serverKey }}"
+{{- end }}
```
Comment (p0lyn0mial) [inline config/kubermatic/templates/dex-grpc-secret.yaml]: some tools and text editors expect a new line at the end, have you tested if this doesn't break helm/deployment ?

Comment (floreks) [reply config/kubermatic/templates/dex-grpc-secret.yaml]: I've been testing it with `helm install --dry-run --debug ...` to make sure that output yaml looks as intended :slightly_smiling_face: I can add a new line here anyway. No problem.

**Pull request merged**